targets/TARGET_ARM_FM/TARGET_FVP_MPS2/us_ticker.c

```diff
@@ -16,77 +16,96 @@
 #include <stddef.h>
 #include "us_ticker_api.h"
 #include "PeripheralNames.h"
-
 #define US_TICKER_TIMER1      CMSDK_DUALTIMER1
 #define US_TICKER_TIMER2      CMSDK_DUALTIMER2
 #define US_TICKER_TIMER_IRQn  DUALTIMER_IRQn
 
+/** mbed OS HAL API defined us_ticker as an increment ticker
+ *  MPS2 platform provided in SSE-200 are decrement tickers
+ *  with interrupt fired counter reaches 0.
+ *
+ *  So 2 Timers are used to construct mbed OS HAL ticker.
+ *  TIMER1 is for counting, and returns inverted binary when read from it
+ *  TIMER2 is for generate interrupts
+ */
 int us_ticker_inited = 0;
 
 void us_ticker_init(void)
 {
     if (us_ticker_inited) {
+        us_ticker_disable_interrupt();
```

Comment:
Is this what targets do? I dont recall seeing it before. init should disable interrupt , but in this case, why only if it was already initialized?

why `us_ticker_inited` variable is not static?

Reply:
Code implements the following ticker requirement:
https://github.com/ARMmbed/mbed-os/blob/ed9a1f1327f5208fd462c5a6938696ee15e393f1/hal/us_ticker_api.h#L64

Reply:
Yes, calling us_ticker_disable_interrupt() on an uninitialized ticker is an [undefined behavior in HAL](https://github.com/ARMmbed/mbed-os/blob/ed9a1f1327f5208fd462c5a6938696ee15e393f1/hal/us_ticker_api.h#L78)
I re-checked, upper level dirvers are not use this variable, it can be staticed, good catch.